### PR TITLE
Heathers bigbranch

### PR DESCRIPTION
**Pull request by hcape**
this branch is lit *dabs*

### CODE REVIEW

File: app/src/main/java/antidose/antidose/MainActivity.java [hunk 1]
```diff
@@ -146,44 +148,75 @@ public void sendAlert(View view) {
...
             return;
         }
 
-        if (location != null && location.getTime() > Calendar.getInstance().getTimeInMillis() - 2 * 60 * 1000) {
+        mLocationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
+        Location locationGPS = mLocationManager.getLastKnownLocation(LocationManager.GPS_PROVIDER);
+        Location locationNETWORK = mLocationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER);
+
+
+        if (locationGPS != null && locationGPS.getTime() > Calendar.getInstance().getTimeInMillis() - 2 * 60 * 1000) {
```
Comment (JakeCooper) [inline app/src/main/java/antidose/antidose/MainActivity.java:166]: I don't know enough about timer math to dispute these magic numbers.

File: app/src/main/java/antidose/antidose/MainActivity.java [hunk 2]
```diff
@@ -146,44 +148,75 @@ public void sendAlert(View view) {
...
+        }
+
+        public void onProviderEnabled(String provider) {
+        }
+
+        public void onStatusChanged(String provider, int status, Bundle extras) {
+        }
+    };
+
+    LocationListener locationListenerNetwork = new LocationListener() {
```
Comment (JakeCooper) [inline app/src/main/java/antidose/antidose/MainActivity.java:204]: Wait why is there two location listeners?

File: app/src/main/java/antidose/antidose/NotifyActivity.java [hunk 3]
```diff
@@ -231,11 +231,19 @@ public void onResponse(Call<RestInterface.IncidentLocation> call, Response<RestI
...
-                            intent.putExtra("incident-latitude", IncidentLatitude);
-                            intent.putExtra("incident-longitude", IncidentLongitude);
+                            if(Math.signum(IncidentLatitude)==0 && Math.signum(IncidentLongitude)==0){
+                                //then it has been cancelled
+                                Intent intent = new Intent(NotifyActivity.this, MainActivity.class);
+                                intent.putExtra("CANCEL_FRAGMENT", "TRUE");
+                                startActivity(intent);
 
-                            startActivity(intent);
+                            }else {
```
Comment (JakeCooper) [inline app/src/main/java/antidose/antidose/NotifyActivity.java:240]: Someone should line up the spacing/indentation at some point, but not for this PR.

**Review by JakeCooper — APPROVED** [review]

**Review by JordanVlieg — APPROVED** [review]
I am happy with these changes.

**Pull request merged**